sql/2019/10_SEO/10_10.sql

```diff
@@ -0,0 +1,21 @@
+#standardSQL
+
+# Linking - extract <a href> count per page (internal + external)
+
+CREATE TEMPORARY FUNCTION parseAnchor(payload STRING, element STRING)
+RETURNS INT64 LANGUAGE js AS '''
+  try {
+    var $ = JSON.parse(payload);
+    var almanac = JSON.parse($._almanac);
+    return almanac['seo-anchor-elements'][element];
+  } catch (e) {
+    return 0;
+  }
+''';
+
+SELECT
+    COUNTIF(parseAnchor(payload, "internal") > 0) as internal,
```

Comment:
A total count of links may not be insightful enough. Generally, absolute counts are less useful out of context of the number of total URLs tested. How about:

- median # of internal links
- median # of external links
- median ratio of internal:external links
- % of pages with more external links
- % of pages with hash links

Not sure what other ways to slice these results might be interesting from an SEO perspective.

Reply:
Good suggestions!